src/main/java/com/aws/greengrass/lifecyclemanager/KernelAlternatives.java

```diff
@@ -277,8 +277,15 @@ public Deployment.DeploymentStage determineDeploymentStage(BootstrapManager boot
      * @throws IOException if file or directory changes fail
      */
     public void activationSucceeds() throws IOException {
-        Path launchDirToCleanUp = Files.readSymbolicLink(getOldDir());
+        Path launchDirToCleanUp = Files.readSymbolicLink(getOldDir()).toAbsolutePath();
         Files.delete(getOldDir());
+        if (Files.isSameFile(launchDirToCleanUp, getCurrentDir())) {
+            logger.atInfo().kv("oldDir", launchDirToCleanUp).log("Skipping launch directory cleanup after kernel "
```

Comment:
nucleus

Reply:
an alternative log message that would be easier for customers to understand would be: "Skipping cleanup of launch directory after Nucleus update as it matches the current launch directory"

The current message invites questions about why the same deployment was executed twice